daemon/algod/api/server/v2/test/handlers_test.go

```diff
@@ -693,21 +696,374 @@ func TestPostTransaction(t *testing.T) {
 	postTransactionTest(t, 0, 200)
 }
 
-func simulateTransactionTest(t *testing.T, txnToUse, expectedCode int, enableTransactionSimulator bool) {
+func simulateTransactionTest(t *testing.T, txnToUse int, format string, expectedCode int, enableTransactionSimulator bool) {
 	handler, c, rec, releasefunc := prepareTransactionTest(t, txnToUse, expectedCode, enableTransactionSimulator)
 	defer releasefunc()
-	err := handler.SimulateTransaction(c)
+	err := handler.SimulateTransaction(c, model.SimulateTransactionParams{Format: (*model.SimulateTransactionParamsFormat)(&format)})
 	require.NoError(t, err)
 	require.Equal(t, expectedCode, rec.Code)
 }
 
+func TestPostSimulateTransaction(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	testCases := []struct {
+		txnIndex                   int
+		format                     string
+		expectedStatus             int
+		enableTransactionSimulator bool
+	}{
+		{
+			txnIndex:                   -1,
+			format:                     "json",
+			expectedStatus:             400,
+			enableTransactionSimulator: true,
+		},
+		{
+			txnIndex:                   0,
+			format:                     "json",
+			expectedStatus:             404,
+			enableTransactionSimulator: false,
+		},
+		{
+			txnIndex:                   0,
+			format:                     "msgpack",
+			expectedStatus:             404,
+			enableTransactionSimulator: false,
+		},
+		{
+			txnIndex:                   0,
+			format:                     "bad format",
+			expectedStatus:             404,
+			enableTransactionSimulator: false,
+		},
+		{
+			txnIndex:                   0,
+			format:                     "json",
+			expectedStatus:             200,
+			enableTransactionSimulator: true,
+		},
+		{
+			txnIndex:                   0,
+			format:                     "msgpack",
+			expectedStatus:             200,
+			enableTransactionSimulator: true,
+		},
+		{
+			txnIndex:                   0,
+			format:                     "bad format",
+			expectedStatus:             400,
+			enableTransactionSimulator: true,
+		},
```

Comment:
format: "format"
also 
format: "type"

Reply:
What new coverage would those cases give us? The actual code to map from string -> encoding happens in a shared helper function, so I don't think this is the best place to put it through its paces.

I chose these the strings "bad format" just to make sure we error in the right way if that shared helper function errors.

Reply:
oh, it would flex the part of the test that calls BindQueryParameters from oapi-codegen, https://github.com/algorand/oapi-codegen/blob/master/pkg/runtime/bindparam.go#L280 , it's fine.